### PR TITLE
Grass walking sfx

### PR DESCRIPTION
**Pull request by Jelmerrr**
prototype grass walk sfx

### CODE REVIEW

**Comment by Jelmerrr** [comment]
Relevante file voor review = Audio/Scripts/GrassWalkingEvent.cs

File: Assets/Audio/Scripts/GrassWalkingEvent.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        if (Physics.Raycast(wolfDownRay, out wolfHit))
+        {
+            wolfDist = wolfHit.distance;
+        }
+    }
+
+    private void UpdateSound() //Updates sound events according to object state
+    {
+        //Check if proper conditions are met for fox
+        if(enable == true && foxZvol <= -0.5 && foxDist <= 0.1 || enable == true && foxZvol >= 0.5 && foxDist <= 0.1)
```
Comment (WinteryFox) [inline Assets/Audio/Scripts/GrassWalkingEvent.cs:75]: Possible bug? Seems you forgot brackets here for the OR condition, should probably be `if ((enable == true && foxZvol <= -0.5 && foxDist <= 0.1) || (enable == true && foxZvol >= 0.5 && foxDist <= 0.1))`

File: Assets/Audio/Scripts/GrassWalkingEvent.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            {
+                //Start event
+                wolfFootstepsGrass.start();
+            }
+        }
+        else
+        {
+            //Stop event
+            wolfFootstepsGrass.stop(STOP_MODE.ALLOWFADEOUT);
+        }
```
Comment (WinteryFox) [inline Assets/Audio/Scripts/GrassWalkingEvent.cs:108]: Duplicated code with line 75-90, make a function instead.

File: Assets/Audio/Scripts/GrassWalkingEvent.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            }
+        }
+        else
+        {
+            //Stop event
+            foxFootstepsGrass.stop(STOP_MODE.ALLOWFADEOUT);
+        }
+
+        //Check if proper conditions are met for wolf
+        if (enable == true && wolfZvol <= -0.5 && wolfDist <= 0.1 || enable == true && wolfZvol >= 0.5 && wolfDist <= 0.1)
```
Comment (WinteryFox) [inline Assets/Audio/Scripts/GrassWalkingEvent.cs:93]: Same possible bug as line 75

File: Assets/Audio/Scripts/GrassWalkingEvent.cs
Comment (WinteryFox) [inline Assets/Audio/Scripts/GrassWalkingEvent.cs:1]: You should probably switch to using `var` over explicit types in your C# code. It's convention and makes for cleaner more readable code.

File: Assets/Audio/Scripts/GrassWalkingEvent.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+        //Execute raycasts
+        if (Physics.Raycast(foxDownRay, out foxHit))
+        {
+            foxDist = foxHit.distance;
+        }
+        if (Physics.Raycast(wolfDownRay, out wolfHit))
+        {
+            wolfDist = wolfHit.distance;
+        }
```
Comment (WinteryFox) [inline Assets/Audio/Scripts/GrassWalkingEvent.cs:69]: Duplicate code, make a function that you call twice for fox and wolf respectively.

File: Assets/Audio/Scripts/GrassWalkingEvent.cs
Comment (WinteryFox) [reply Assets/Audio/Scripts/GrassWalkingEvent.cs:1]: You have a lot of duplicate code which makes maintainability harder and can introduce copy-paste bugs. Make sure to pay attention to this in the future.

**Review by WinteryFox — CHANGES_REQUESTED** [review]

**Review by WinteryFox — APPROVED** [review]

**Pull request merged**